### ISSUE TITLE
Error trying to use polars to load a parquet file written by polars

### ISSUE THREAD

**Issue by mkleinbort-ic** [pola-rs/polars#14898]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

This is a bit hard to reproduce - I'll spend some time trying to generate a minimal file, but here is the error:

```python

df.write_parquet('test.parquet')
pl.read_parquet('test.parquet')

>>> ComputeError: not implemented: dictionary arrays from non-dict-encoded pages

```

### Issue description

I am able to read a parquet file, filter it, and save it to parquet.

But I can't read the filtered file back from parquet.

I'll figure out what data type is causing the issue

### Expected behavior

Should be able to read the parquet file written by polars.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.13
Index type:           UInt32
Platform:             Windows-10-10.0.22621-SP0
Python:               3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  0.10.0
cloudpickle:          3.0.0
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.12.2
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.8.3
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              15.0.0
pydantic:             2.6.1
pyiceberg:            0.6.0
pyxlsb:               <not installed>
sqlalchemy:           2.0.28
xlsx2csv:             0.8.2
xlsxwriter:           3.2.0
```

</details>

**Comment by mkleinbort-ic** [comment]
Ok, narrowed it down to a column of schema:

```python
OrderedDict([('column', Struct({'x1: List(Categorical(ordering='physical')), 'x2': List(Categorical(ordering='physical'))}))])
```

Also the original dataframe (pre-filter) was 30m+ rows, and the one after filtering is 350k rows - in case it's related to a parquet row group size.

Still struggling to reproduce the issue with a minimal file.

**Comment by mkleinbort-ic** [comment]
Partial reproduction of the error:

With: 

```python
from itertools import cycle

c1 = cycle(['A','B', 'C', None])
c2 = cycle(['A','B', 'C','D',  None])
df = pl.DataFrame({
    'x1': [next(c1) for i in range(300_000)],
    'x2': [next(c2) for i in range(300_000)],
}).cast(pl.Categorical(ordering='physical')).with_columns(
    xs = pl.concat_list('x1', 'x2')
)
```

This works (50ms):
```python
df.write_parquet('test.parquet')
pl.read_parquet('test.parquet')
```

This runs for a long time - I got tired of waiting after 10 minutes so not sure if it will ever finish.

```python
df.tail(1).write_parquet('test.parquet')
pl.read_parquet('test.parquet')
```

**Comment by mkleinbort-ic** [comment]
Issue also in `polars = 0.20.14`

**Comment by cmdlineluser** [comment]
The partial repro hangs on the `.write_parquet()` for me.

```python
df.tail(1)
# shape: (1, 3)
# ┌──────┬──────┬──────────────┐
# │ x1   ┆ x2   ┆ xs           │
# │ ---  ┆ ---  ┆ ---          │
# │ cat  ┆ cat  ┆ list[cat]    │
# ╞══════╪══════╪══════════════╡
# │ null ┆ null ┆ [null, null] │
# └──────┴──────┴──────────────┘
```

Just experimenting with some other write methods, it panics for json/ndjson:

```python
df.tail(1).write_ndjson()
# thread '<unnamed>' panicked at crates/polars-json/src/json/write/serialize.rs:233:46:
# called `Option::unwrap()` on a `None` value
# PanicException: called `Option::unwrap()` on a `None` value
```

**Comment by mkleinbort-ic** [comment]
More info:

I still can't reproduce the issue with a clean dataframe, but this is interesting:

As mentioned, the problem column is a Struct of lists of categoricals...

```python
df.select('problemColumn').write_parquet('test.parquet')
pl.read_parquet('test.parquet') # <--- this fails with error: ComputeError: not implemented: dictionary arrays from non-dict-encoded pages

```

But this works:
```python
df.select('problemColumn').unnest('problemColumn').write_parquet('test.parquet')
pl.read_parquet('test.parquet')
```

**Comment by ritchie46** [comment]
I think this is fixed in main, but cannot check it because there is no repro.

**Comment by mkleinbort** [comment]
I'll check it when the next version comes out.

**Comment by ritchie46** [comment]
Can we close this one @mkleinbort ?

**Comment by mkleinbort-ic** [comment]
Fixed in the version 0.20.15
